activitypub-core/src/main/java/org/xwiki/contrib/activitypub/internal/json/AbstractActivityPubObjectReferenceDeserializer.java

```diff
@@ -48,15 +47,24 @@
 {
     @Override
     public ActivityPubObjectReference<ActivityPubObject> deserialize(JsonParser jsonParser,
-        DeserializationContext deserializationContext) throws IOException, JsonProcessingException
+        DeserializationContext deserializationContext) throws IOException
     {
         ObjectMapper mapper = (ObjectMapper) jsonParser.getCodec();
 
-        ActivityPubObjectReference<ActivityPubObject> objectReference = new ActivityPubObjectReference<>();
+        ActivityPubObjectReference<ActivityPubObject> objectReference;
         if (jsonParser.currentToken() == JsonToken.START_OBJECT) {
-            objectReference.setObject(mapper.readValue(jsonParser, ActivityPubObject.class));
+            // If reading the contained object returns null, we skip the reference creation and return the null value 
+            // too.
+            ActivityPubObject object = mapper.readValue(jsonParser, ActivityPubObject.class);
+            if (object != null) {
+                objectReference = new ActivityPubObjectReference<>();
+                objectReference.setObject(object);
+            } else {
+                objectReference = null;
```

Comment:
I'm pretty sure that our javadoc did not express that the returns of this method could be null. To be checked.

Reply:
Actually it's not an interface we defined. In that case we should probably add some javadoc here to clarify.